### DISCUSSION TITLE
Console errors with Alpine JS - Webpack 4

### DISCUSSION THREAD

**Discussion by gAusWeb** [alpinejs/alpine#3704] [7. Bugs (archived)] [answered]
Getting console errors when click functionality triggered:

**Error: on each click, this error appears 5 times**
Uncaught TypeError: Cannot read properties of undefined (reading 'contains')
    at eval (eval at safeAsyncFunction (module.cjs.js:1740:1), <anonymous>:3:46)

**Environment:**
devDependencies
"webpack": "^4.35.3", 
"webpack-cli": "^3.1.0",

dependencies:
"@alpinejs/collapse": "^3.12.3",
"alpinejs": "^3.12.3",

**Alpine setup:**
```
import Alpine from "alpinejs/dist/module.cjs";
import collapse from "@alpinejs/collapse";

window.Alpine = Alpine;
 Alpine.plugin(collapse);
 Alpine.start();
```

**Demo usage:**

```
<div x-data="{myTest: false}">
  <div x-show="myTest" x-collapse.min.50px>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Deserunt excepturi alias praesentium, error reiciendis architecto nobis. Ad quasi ipsam provident neque at, excepturi eum? Officia quasi corporis tempora minus quia itaque distinctio dolorum debitis maxime. Dolorum tempora aspernatur numquam, modi nemo ad laboriosam rerum vero nihil explicabo quasi distinctio fugit. Lorem ipsum dolor sit amet consectetur adipisicing elit. Deserunt excepturi alias praesentium, error reiciendis architecto nobis. Ad quasi ipsam provident neque at, excepturi eum? Officia quasi corporis tempora minus quia itaque distinctio dolorum debitis maxime. Dolorum tempora aspernatur numquam, modi nemo ad laboriosam rerum vero nihil explicabo quasi distinctio fugit.
  </div>
  <button @click="myTest = !myTest">test me</button>
</div>
```

Any ideas on how to stop these errors without updating to Webpack 5, and thanks in advance...?

**Comment by ekwoka** [comment, marked as answer]
Can you verify this doesn't have anything to do with x-collapse?

x-collapse shouldn't use .contains.

But it seems like something is causing the element to not be properly passed through somewhere...but it shouldn't need to be.

Are you 100% positive it's this here?

What does it say the code is for the eval?

Nothing in this code should warrant this error happening in the eval.

**Comment by gAusWeb** [comment]
OMG, thank-you so much!! I stripped back my code to only have the x-collapse example shown above (as i did indeed have quite a lot of Alpine functionality at work) and indeed the error disappeared. 

The culprit was the dropdown menus. I was calling upon a x-ref, but had not declared the x-ref's within the correct scope. The error was multiplied 5 times because I had 5 dropdown menus. 

The problem was indeed me and not this lovely plugin - my sincere apologies! 

Thank you so much, as you just saved me from spending copious amounts of late nights, writing custom code to meet my deadline.

Ticket Status: Resolved!
